packages/public/common-ui-web/src/components/FullscreenImageModal/FullscreenImageModal.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import { FullscreenModal } from '../FullscreenModal';
+import { styles } from './FullscreenImageModal.styles';
+
+/**
+ * Props for the FullscreenImageModal component.
+ */
+export interface FullscreenImageModalProps {
+  show: boolean;
+  label?: string;
+  onClose?: () => void;
+  url: string;
+}
+
+function calculatePosition(
+  viewPort: number,
+  imageDimension: number,
+  clickPosition: number,
+  scale: number,
+) {
+  if (viewPort > imageDimension * 3) {
+    return 0;
+  }
+  const blackBand = (viewPort - imageDimension) / 2;
+  const maxPosition = (imageDimension - blackBand) / scale;
+  return Math.min(maxPosition, Math.max(-maxPosition, imageDimension / 2 - clickPosition));
+}
+
+/**
+ * FullscreenImageModal component used to display a full-screen modal for an image and able the user to zoom on it.
+ */
+export function FullscreenImageModal({
+  show,
+  label = '',
+  onClose,
+  url,
+}: FullscreenImageModalProps) {
+  const [isZoomed, setIsZoomed] = useState(false);
+  const [position, setPosition] = useState({ x: 0, y: 0 });
+  const scale = 3;
```

Comment:
If this constant is not reactive, put it outside the component so that it is not redeclared at every render. Also : if it is a constant, name it in SCREAMING_CASE